synapse/handlers/register.py

```diff
@@ -513,6 +513,34 @@ def get_or_register_3pid_guest(self, medium, address, inviter_user_id):
 
     @defer.inlineCallbacks
     def _join_user_to_room(self, requester, room_identifier):
+
+        # try to create the room if we're the first user on the server
+        if self.hs.config.autocreate_auto_join_rooms:
+            count = yield self.store.count_all_users()
+            if count == 1 and RoomAlias.is_valid(room_identifier):
```

Comment:
it feels very random to be doing this at this point. (Which is when the first user is registered? when they log in?). what if two users register at once? what if the first user didn't qualify for autojoining for some reason?

Can we not do this when the HS first starts, somehow?

Reply:
the only way to do it at start would be if we also autocreated an admin user, which in turn then means shared-secret registering them and then setting up their profile and 3pid and passwords etc correctly.  agreed this would be cleaner, but this is intended as a fast fix to avoid having to do that in the provisioning process, on the basis that the first user who signs in is 99% always going to be the server admin, and if it isn't, then you can always repoint the alias.